### PR TITLE
Styling cleanup

### PR DESCRIPTION
**Pull request by mhevery**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/render3/instructions/styling.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ export function ɵɵstyleMap(
  * Parse text as style and add values to ArrayMap.
  *
  * This code is pulled out to a separate function so that it can be tree shaken away if it is not
- * needed. It is only reference from `ɵɵstyleMap`.
+ * needed. It is only referenced from `ɵɵstyleMap`.
  *
  * @param arrayMap ArrayMap to add parsed values to.
```
Comment (kara) [inline packages/core/src/render3/instructions/styling.ts]: Can we rename `ArrayMap` as suggested here

https://github.com/angular/angular/pull/34804#discussion_r370883041

File: packages/core/src/render3/instructions/styling.ts [hunk 2]
```diff
@@ -729,6 +742,7 @@ function updateStyling(
 function findStylingValue(
     tData: TData, tNode: TNode | null, lView: LView, prop: string, index: number,
     isClassBased: boolean): any {
+  const isPrevDirection = tNode === null;
```
Comment (kara) [inline packages/core/src/render3/instructions/styling.ts]: I think the comment on lines 732 - 736 belongs here. (`- `TNode` search next and quit as soon as `isStylingValuePresent(value)` is true.` etc)

File: packages/core/src/render3/instructions/styling.ts [hunk 3]
```diff
@@ -219,12 +219,16 @@ export function checkStylingProperty(
...
-* @param tStylingMapKey See `STYLE_MAP_STYLING_KEY` and `CLASS_MAP_STYLING_KEY`.
-* @param value binding value.
-* @param isClassBased `true` if `class` change (`false` if `style`)
-*/
+ * Common code between `ɵɵclassMap` and `ɵɵstyleMap`.
+ *
+ * @param arrayMapSet (See `arrayMapSet` in "util/array_utils") Gets passed in as a function so that
+ *        `style` can pass in version which does sanitization. This is done for tree shaking
+ *        purposes.
+ * @param stringParser Parser used to pares `value` if `string`. (Passed in as `style` and `class`
```
Comment (kara) [inline packages/core/src/render3/instructions/styling.ts]:
```suggestion
 * @param stringParser Parser used to parse `value` if `string`. (Passed in as `style` and `class`
```

**Review by kara — CHANGES_REQUESTED** [review]
What is here looks good, but seems like some comments were missed, for ex:

https://github.com/angular/angular/pull/34804#discussion_r370900249
https://github.com/angular/angular/pull/34804#discussion_r370897939
https://github.com/angular/angular/pull/34804#discussion_r370900759

File: packages/core/src/render3/instructions/styling.ts (hunk 1 above)
Comment (mhevery) [reply packages/core/src/render3/instructions/styling.ts]: Done

File: packages/core/src/render3/instructions/styling.ts (hunk 3 above)
Comment (mhevery) [reply packages/core/src/render3/instructions/styling.ts]: fixed

File: packages/core/src/render3/instructions/styling.ts (hunk 2 above)
Comment (mhevery) [reply packages/core/src/render3/instructions/styling.ts]: fixed

File: packages/core/src/render3/state.ts [hunk 4]
```diff
@@ -240,17 +240,6 @@ export function getLView(): LView {
...
- * (when the provided elementIndex value has changed).
- *
- * @param elementIndex the element index value for the host element where
- *                     the directive/component instance lives
- */
-export function setActiveHostElement(elementIndex: number) {
-  setSelectedIndex(elementIndex);
-}
-
 export function clearActiveHostElement() {
```
[the remark this answers is not in the document]
Comment (mhevery) [reply packages/core/src/render3/state.ts]: Sorry about that, it is gone now.

File: packages/core/src/render3/state.ts [hunk 5]
```diff
@@ -240,17 +240,6 @@ export function getLView(): LView {
...
   setSelectedIndex(-1);
 }
```
[the remark this answers is not in the document]
Comment (mhevery) [reply packages/core/src/render3/state.ts]: Sorry, good catch!

File: packages/core/src/render3/state.ts [hunk 6]
```diff
@@ -332,11 +324,22 @@ export function incrementBindingIndex(count: number): number {
  * Bindings inside the host template are 0 index. But because we don't know ahead of time
  * how many host bindings we have we can't pre-compute them. For this reason they are all
  * 0 index and we just shift the root so that they match next available location in the LView.
- * @param value
+ * @param bindingRootIndex
  */
-export function setBindingRootForHostBindings(value: number) {
-  const lframe = instructionState.lFrame;
-  lframe.bindingIndex = lframe.bindingRootIndex = value;
+export function setBindingRootForHostBindings(
```
Comment (diicar) [inline packages/core/src/render3/state.ts]: missing `@param currentDirectiveIndex`

File: packages/core/src/render3/instructions/styling.ts [hunk 7]
```diff
@@ -127,17 +127,17 @@ export function ɵɵstyleMap(
...
  *
- * @param arrayMap ArrayMap to add parsed values to.
+ * @param keyValueArray KeyValueArray to add parsed values to.
  * @param text text to parse.
  */
-export function styleStringParser(arrayMap: ArrayMap<any>, text: string): void {
+export function styleStringParser(keyValueArray: KeyValueArray<any>, text: string): void {
   for (let i = parseStyle(text); i >= 0; i = parseStyleNext(text, i)) {
-    styleArrayMapSet(arrayMap, getLastParsedKey(text), getLastParsedValue(text));
+    styleArrayMapSet(keyValueArray, getLastParsedKey(text), getLastParsedValue(text));
```
Comment (kara) [inline packages/core/src/render3/instructions/styling.ts]: I think this method needs to be `styleKeyValueArraySet`

```suggestion
    styleKeyValueArraySet(keyValueArray, getLastParsedKey(text), getLastParsedValue(text));
```

File: packages/core/src/render3/instructions/styling.ts [hunk 8]
```diff
@@ -528,50 +531,53 @@ export function getHostDirectiveDef(tData: TData): DirectiveDef<any>|null {
...
 export function toStylingArrayMap(
-    arrayMapSet: (arrayMap: ArrayMap<any>, key: string, value: any) => void,
-    stringParser: (styleArrayMap: ArrayMap<any>, text: string) => void, value: string|string[]|
-    {[key: string]: any}|Map<any, any>|Set<any>|null|undefined): ArrayMap<any> {
+    keyValueArraySet: (keyValueArray: KeyValueArray<any>, key: string, value: any) => void,
+    stringParser: (styleArrayMap: KeyValueArray<any>, text: string) => void, value: string|string[]|
+    {[key: string]: any}|Map<any, any>|Set<any>|null|undefined): KeyValueArray<any> {
   if (value == null /*|| value === undefined */ || value === '') return EMPTY_ARRAY as any;
-  const styleArrayMap: ArrayMap<any> = [] as any;
+  const styleArrayMap: KeyValueArray<any> = [] as any;
```
Comment (kara) [inline packages/core/src/render3/instructions/styling.ts]:
```suggestion
  const styleKeyValueArray: KeyValueArray<any> = [] as any;
```

File: packages/core/src/render3/instructions/styling.ts [hunk 9]
```diff
@@ -605,23 +611,24 @@ function styleArrayMapSet(arrayMap: ArrayMap<any>, key: string, value: any) {
...
+ * very
  * cheap to compute deltas between the previous and current value.
  *
  * @param tView Associated `TView.data` contains the linked list of binding priorities.
  * @param tNode `TNode` where the binding is located.
  * @param lView `LView` contains the values associated with other styling binding at this `TNode`.
  * @param renderer Renderer to use if any updates.
- * @param oldArrayMap Previous value represented as `ArrayMap`
- * @param newArrayMap Current value represented as `ArrayMap`
+ * @param oldArrayMap Previous value represented as `KeyValueArray`
```
Comment (kara) [inline packages/core/src/render3/instructions/styling.ts]:
```suggestion
 * @param oldKeyValueArray Previous value represented as `KeyValueArray`
```

File: packages/core/src/render3/instructions/styling.ts [hunk 10]
```diff
@@ -605,23 +611,24 @@ function styleArrayMapSet(arrayMap: ArrayMap<any>, key: string, value: any) {
...
+ * @param newArrayMap Current value represented as `KeyValueArray`
```
Comment (kara) [inline packages/core/src/render3/instructions/styling.ts]:
```suggestion
 * @param newKeyValueArray Current value represented as `KeyValueArray`
```

File: packages/core/src/render3/interfaces/node.ts [hunk 11]
```diff
@@ -498,7 +498,7 @@ export interface TNode {
   styles: string|null;
 
   /**
-   * An `ArrayMap` version of residual `styles`.
+   * An `KeyValueArray` version of residual `styles`.
```
Comment (kara) [inline packages/core/src/render3/interfaces/node.ts]:
```suggestion
   * A `KeyValueArray` version of residual `styles`.
```

File: packages/core/src/render3/interfaces/node.ts [hunk 12]
```diff
@@ -536,15 +536,15 @@ export interface TNode {
   classes: string|null;
 
   /**
-   * An `ArrayMap` version of residual `classes`.
+   * An `KeyValueArray` version of residual `classes`.
```
Comment (kara) [inline packages/core/src/render3/interfaces/node.ts]:
```suggestion
   * A `KeyValueArray` version of residual `classes`.
```

File: packages/core/src/render3/styling/style_binding_list.ts [hunk 13]
```diff
@@ -204,10 +204,10 @@ export function insertTStylingBinding(
   let tStylingKey: TStylingKeyPrimitive;
   if (Array.isArray(tStylingKeyWithStatic)) {
     // We are case when the `TStylingKey` contains static fields as well.
-    const staticArrayMap = tStylingKeyWithStatic as ArrayMap<any>;
+    const staticArrayMap = tStylingKeyWithStatic as KeyValueArray<any>;
```
Comment (kara) [inline packages/core/src/render3/styling/style_binding_list.ts]:
```suggestion
    const staticKeyValueArray = tStylingKeyWithStatic as KeyValueArray<any>;
```

File: packages/core/test/render3/instructions/styling_spec.ts [hunk 14]
```diff
@@ -403,51 +403,51 @@ describe('styling', () => {
...
   describe('toStylingArray', () => {
     describe('falsy', () => {
-      it('should return empty ArrayMap', () => {
-        expect(toStylingArrayMap(arrayMapSet, null !, '')).toEqual([] as any);
-        expect(toStylingArrayMap(arrayMapSet, null !, null)).toEqual([] as any);
-        expect(toStylingArrayMap(arrayMapSet, null !, undefined)).toEqual([] as any);
-        expect(toStylingArrayMap(arrayMapSet, null !, [])).toEqual([] as any);
-        expect(toStylingArrayMap(arrayMapSet, null !, {})).toEqual([] as any);
+      it('should return empty KeyValueArray', () => {
+        expect(toStylingArrayMap(keyValueArraySet, null !, '')).toEqual([] as any);
```
Comment (kara) [inline packages/core/test/render3/instructions/styling_spec.ts]: Seems like we need to rename `toStylingArrayMap` as well

```suggestion
        expect(toStylingKeyValueArray(keyValueArraySet, null !, '')).toEqual([] as any);
```

File: packages/core/src/render3/interfaces/node.ts [hunk 15]
```diff
@@ -536,15 +536,15 @@ export interface TNode {
...
    *
    * Same as `TNode.residualStyles` but for classes.
    *
    * - `undefined': not initialized.
    * - `null`: initialized but `classes` is `null`
-   * - `ArrayMap`: parsed version of `S`.
+   * - `KeyValueArray`: parsed version of `S`.
```
Comment (kara) [inline packages/core/src/render3/interfaces/node.ts]: Looks like the comments from the PR description still need to be addressed

https://github.com/angular/angular/pull/35022#pullrequestreview-349823025

**Review by kara — CHANGES_REQUESTED** [review]
A few more updates to make to clean up `ArrayMap`

**Review by kara — APPROVED** [review]
LGTM

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:292212730:BASE:292212806:1580335170651:aadab30d)

**Pull request closed without merging**